django_mongodb/schema.py

```diff
@@ -60,21 +96,97 @@ def remove_field(self, model, field):
+        if index.contains_expressions:
+            return
+        index_orders = (
+            [(field.column, 1)]
+            if field
+            else [
+                (model._meta.get_field(field_name).column, 1 if order == "" else -1)
+                for field_name, order in index.fields_orders
+            ]
+        )
+        idx = IndexModel(index_orders, name=index.name)
+        self.connection.database[model._meta.db_table].create_indexes([idx])
 
-    def rename_index(self, model, old_index, new_index):
-        pass
+    def _add_composed_index(self, model, field_names):
+        """Add an index on the given list of field_names."""
+        idx = Index(fields=field_names)
+        idx.set_name_with_model(model)
+        self.add_index(model, idx)
+
+    def _add_field_index(self, model, field):
+        """Add an index on a field with db_index=True."""
+        index = Index(fields=[field.name])
+        index.name = self._create_index_name(model._meta.db_table, [field.column])
+        self.add_index(model, index, field=field)
 
     def remove_index(self, model, index):
-        pass
+        if index.contains_expressions:
+            return
+        self.connection.database[model._meta.db_table].drop_index(index.name)
+
+    def _remove_composed_index(self, model, field_names, constraint_kwargs):
+        """
+        Remove the index on the given list of field_names created by
+        index/unique_together, depending on constraint_kwargs.
+        """
+        meta_constraint_names = {constraint.name for constraint in model._meta.constraints}
+        meta_index_names = {constraint.name for constraint in model._meta.indexes}
+        columns = [model._meta.get_field(field).column for field in field_names]
+        constraint_names = self._constraint_names(
+            model,
+            columns,
+            exclude=meta_constraint_names | meta_index_names,
+            **constraint_kwargs,
+        )
+        if len(constraint_names) != 1:
+            num_found = len(constraint_names)
+            columns_str = ", ".join(columns)
+            raise ValueError(
+                f"Found wrong number ({num_found}) of constraints for "
+                f"{model._meta.db_table}({columns_str})."
+            )
+        collection = self.connection.database[model._meta.db_table]
+        collection.drop_index(constraint_names[0])
+
+    def _remove_field_index(self, model, field):
+        """Remove a field's db_index=True index."""
+        collection = self.connection.database[model._meta.db_table]
+        meta_index_names = {index.name for index in model._meta.indexes}
+        index_names = self._constraint_names(
+            model,
+            [field.column],
+            index=True,
+            # Retrieve only BTREE indexes since this is what's created with
+            # db_index=True.
+            type_=Index.suffix,
+            exclude=meta_index_names,
+        )
```

Comment:
why do we need to use index_names. Is it because there will be special prefixes attached to the naming on django-based indexes?

Reply:
The index name contains `model._meta.db_table` and if that changes after the index was created, the index isn't regenerated with a new name.